iothub/device/src/Edge/EdgeModuleClientFactory.cs

```diff
@@ -92,13 +92,14 @@ private async Task<IotHubModuleClient> CreateInternalClientFromEnvironmentAsync(
 
                 ISignatureProvider signatureProvider = new HttpHsmSignatureProvider(edgedUri, DefaultApiVersion);
 
-                TimeSpan sasTokenTimeToLive = _options.SasTokenTimeToLive;
-                int sasTokenRenewalBuffer = _options.SasTokenRenewalBuffer;
-
-#pragma warning disable CA2000 // Dispose objects before losing scope - IDisposable ModuleAuthenticationWithHsm is disposed when the client is disposed.
-                // Since the sdk creates the instance of disposable ModuleAuthenticationWithHsm, the sdk needs to dispose it once the client is disposed.
-                var authMethod = new ModuleAuthenticationWithHsm(signatureProvider, deviceId, moduleId, generationId, sasTokenTimeToLive, sasTokenRenewalBuffer);
-#pragma warning restore CA2000 // Dispose objects before losing scope - IDisposable ModuleAuthenticationWithHsm is disposed when the client is disposed.
+                // TODO: environment variables need to be added to accept SasTokenTimeToLive and SasTokenRenewalBuffer.
```

Comment:
This is an open #TODO item

Reply:
Is there any reason why you want to take them as environment variables rather than in ClientOptions? Are these env vars set up for the user in Edge scenarios?

Reply:
I concur. As a rule, I don't like using environment variables for client library input, aside from the Azure Edge case.

Reply:
The environmental variable is intended to be for Azure Edge case only. That is the only scenario where ModuleAuthenticationWithHsm is used. This is the internal IAuthenticationMethod the SDK creates for you when the user initializes a ModuleClient through the "CreateFromEnvironmentAsync" flow. 
Since we ask the user to provide all other credential params through environment variables, i.e. iot hub name, device Id, module Id, edge URI Id, etc.), it seems reasonable to me to have them pass the sas ttl through env variables as well. That said, I do mean to confirm with the Edge team before introducing any additional env variables.